splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy

```diff
@@ -36,25 +37,33 @@ public class UserActionDSL {
                         env: buildParameters, listener: listener);
                 Binding binding = new Binding();
                 binding.setVariable("splunkins", delegate);
-                ClassLoader cl = Jenkins.getActiveInstance().getPluginManager().uberClassLoader;
+                binding.setVariable("build", build);
+                binding.setVariable("env", buildParameters);
                 // the script written before SecureGroovyScript was introduced, and can not be migrated automatically
                 boolean legacyScript = SplunkJenkinsInstallation.get().isLegacyMode();
                 try {
+                    ApprovalContext context = ApprovalContext.create().withCurrentUser().withItemAsKey(build.getParent());
+                    //saving pending for approval
+                    String groovyScriptText = ScriptApproval.get().configuring(scriptText, GroovyLanguage.get(), context);
```

Comment:
Script maybe loaded from a file stored on Jenkins server, the script path is configured by user, but the script content will change after configuration.